Get Refresh History and Data Sources

addahlin opened the pull request:
I added functions to the datasets.py file to retrieve the refresh history and to get the datasources associated with the Dataset.
I also created a static function in client.py using default values for authentication; I believe this is what most people will use and it makes it much simpler to use.
Finally, I added a util.py file with convenience functions to convert the datetime strings retrieved from PowerBI into Python datetime objects.

cmberryau commented on pypowerbi/client.py:
```diff
@@ -12,10 +13,34 @@
     api_myorg_snippet = 'myorg'
 
+    @staticmethod
+    def get_client(client_id, username, password, authority_url=None, resource_url=None, api_url=None):
```
Good idea with the default get_client method, but I think it should be named get_client_with_username_password so it is cohesive with adal.AuthenticationContext.acquire_token_with_username_password naming. Documentation also would be suitable.

addahlin replied:
Good ideas! Should have added tests and docs in the first place. Will add them and update.

popeus commented:
@addahlin do you intend to write the tests and finish this PR?

addahlin commented:
> @addahlin do you intend to write the tests and finish this PR?

@popeus That's the plan, it's just been perpetually on the back burner. I'll focus on getting this buttoned up in the next week or two though.

addahlin commented:
@cmberryau I've added documentation and unit tests. I also renamed the get_client() function as requested.
I couldn't figure out how to get the client unit tests working to add tests for the refresh_dataset() and get_dataset_refresh_history() functions. Is there a mock or do you have a Power BI instance you're hittng?

cmberryau commented:
For tests that actually need to touch the PowerBI API I am just using a blank PowerBI instance for testing those. I can think about a way to best allow other contributors to use this or instructions on how to set up their own. Do you have suggestions? Also, thanks for the work on the PR - I will merge this weekend.

addahlin commented:
@cmberryau Other than making a mock for the Power BI Service, not sure. Does Microsoft have a publicly accessible instance anyone can hit? If so, maybe that could be used so that a Power BI license isn't required... although not sure why anyone would use this library if they don't already have a license.

cmberryau commented:
@addahlin interesting idea - I don't think so re a public instance, but it's worth asking
